pyomo/contrib/doe/fim_doe_tutorial.ipynb

```diff
@@ -0,0 +1,2017 @@
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "fixed = {}\n",
+    "all_fim.figure_drawing(fixed, ['CA0','T'], 'Reactor case','$C_{A0}$ [M]', 'T [K]' )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1a54b6e6",
+   "metadata": {},
+   "source": [
+    "## Grid search for 3 design variables"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "306eb485",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "# Define design ranges\n",
+    "design_ranges = [list(np.linspace(1,5,2)),  list(np.linspace(300,700,2)), [300,500]]\n",
+    "\n",
+    "# Define design variable \n",
+    "# Here the two T are for different controlling time subsets\n",
+    "dv_apply_name = ['CA0', 'T', 'T']\n",
+    "dv_apply_time = [[0], [0], [0.125, 0.25, 0.375, 0.5, 0.625, 0.75, 0.875,1]]\n",
+    "\n",
+    "# Define experiments\n",
+    "exp1 = generate_exp(t_control, 5, [300, 300, 300, 300, 300, 300, 300, 300, 300])\n",
+    "\n",
+    "## choose from 'sequential_finite', 'direct_kaug'\n",
+    "#sensi_opt = 'sequential_finite'\n",
+    "sensi_opt = 'direct_kaug'\n",
+    "\n",
+    "# model option\n",
+    "if sensi_opt == 'direct_kaug':\n",
+    "    args_ = [False]\n",
+    "else:\n",
+    "    args_ = [True]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "167d2800",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "doe_object = DesignOfExperiments(parameter_dict, dv_pass,\n",
+    "                                 measure_class, createmod,\n",
+    "                                prior_FIM=prior_pass, discretize_model=disc, args=args_)\n",
+    "\n",
+    "all_fim = doe_object.run_grid_search(exp1, design_ranges, dv_apply_name, dv_apply_time, \n",
+    "                                     mode=sensi_opt)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "73704fec",
+   "metadata": {},
+   "source": [
+    "### Draw 1D sensitivity curve"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "58a96488",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "test = all_fim.extract_criteria()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e26fa95d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## draw 1D sensitivity curve \n",
+    "\n",
+    "fixed = {\"'CA0'\": 1.0, \"'T2'\": 300}\n",
+    "\n",
+    "all_fim.figure_drawing(fixed, ['T'], 'Reactor case','T [K]','$C_{A0}$ [M]' )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e215ba75",
+   "metadata": {},
+   "source": [
+    "### Draw 2D sensitivity curve"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "a46ee3d2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fixed = {\"'T2'\": 300}\n",
+    "\n",
+    "all_fim.figure_drawing(fixed, ['CA0','T'], 'Reactor case','$C_{A0}$ [M]', 'T [K]' )"
```

Comment:
Could you add images to the notebook with the figures you expect to be generated by this code?